moses/atomese/representation/atomese_representation.cc

```diff
@@ -0,0 +1,449 @@
+				cells.push_back(mk_real_cell(cell_value));
+
+			Handle h = createLink(LIST_LINK,
+			                      mk_row_number_cell(++row_num),
+			                      createLink(cells, LIST_LINK));
+			rows.push_back(h);
+		}
+	}
+
+	return createLink(SIMILARITY_LINK, parse_header(labels),
+	                  createLink(rows, SET_LINK));
+}
+
+/**
+ * Represent the table using either similarity or list link.
+ *
+ * @param table
+ * @param similarity use similarity link if true
+ * @return
+ */
+HandleSeq mk_full_table(const Table& table, const bool similarity=false)
+{
+	const type_node type = table.otable.get_type();
+	HandleSeq rows;
+	int row_num = 0;
+
+	if (type == id::boolean_type)
+	{
+		for (const multi_type_seq& m : table.itable)
+		{
+			HandleSeq cells;
+
+			cells.push_back(mk_boolean_cell(
+				boost::get<builtin>(table.otable[row_num])));
+
+			for (builtin cell_value : m.get_seq<builtin>())
+				cells.push_back(mk_boolean_cell(cell_value));
+
+			rows.push_back(
+				similarity ? mk_numbered_row(cells, row_num + 1)
+				           : mk_row(cells));
+			row_num++;
+		}
+	}
+	else if (type == id::contin_type)
+	{
+		for (const multi_type_seq& m : table.itable)
+		{
+			HandleSeq cells;
+
+			cells.push_back(mk_real_cell(
+				boost::get<contin_t>(table.otable[row_num])));
+
+			for (contin_t cell_value : m.get_seq<contin_t>())
+				cells.push_back(mk_real_cell(cell_value));
+
+			rows.push_back(
+				similarity ? mk_numbered_row(cells, row_num + 1)
+				           : mk_row(cells));
+			row_num++;
+		}
+	}
+
+	return rows;
+}
+
+/**
+ * Represent a table without using any compression.
+ *
+ * @param table
+ * @param use_eval use EvaluationLink for boolean table representation.
+ * @return
+ */
+Handle mk_unfolded_table(const Table& table, const bool use_eval)
+{
+	HandleSeq rows;
+	int row_num = 0;
+	const type_node type = table.otable.get_type();
+	const vector<string>& labels = table.itable.get_labels();
+
+	if (type == id::boolean_type)
+	{
+		if (use_eval)
+		{
+			for (const multi_type_seq& m : table.itable)
+			{
+				HandleSeq cells;
+				int cell_number = 0;
+
+				cells.push_back(mk_boolean_cell_eval(
+					boost::get<builtin>(table.otable[row_num]),
+					row_num + 1,
+					table.otable.get_label()));
+
+				for (builtin cell_value : m.get_seq<builtin>())
+					cells.push_back(
+						mk_boolean_cell_eval(cell_value, row_num + 1,
+						                     labels[cell_number++]));
+
+				rows.push_back(mk_row(cells));
+				row_num++;
+			}
+		}
+		else
+		{
+			for (const multi_type_seq& m : table.itable)
+			{
+				HandleSeq cells;
+				int cell_number = 0;
+
+				cells.push_back(mk_boolean_cell_exec(
+					boost::get<builtin>(table.otable[row_num]),
+					row_num + 1,
+					table.otable.get_label()));
+
+				for (builtin cell_value : m.get_seq<builtin>())
+					cells.push_back(
+						mk_boolean_cell_exec(cell_value, row_num + 1,
+						                     labels[cell_number++]));
```

Comment:
Hmm, yes, I feel that if you wrap that code in a visitor you may simply a lot your code. Maybe like having a visitor for turning a `multi_type_seq` into a `cells` vector.